### PR TITLE
Backport! XSI-894

### PR DESCRIPTION
**Pull request by lippirk**
Backport to Stockholm of:
 - https://github.com/xapi-project/stdext/pull/58
 - https://github.com/xapi-project/stdext/pull/56
(both are necessary)

### CODE REVIEW

**Comment by lippirk** [comment]
When this is merged I'll made a PR to xs-opam

**Review by psafont — APPROVED** [review]

**Comment by psafont** [comment]
Note that tests are not being run by travis. See https://github.com/xapi-project/stdext/commit/10edcbabedeadf68d531b15d8ef11309d379351f#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485 and https://github.com/xapi-project/stdext/commit/79fc7075200c10aa80ed93c5462c344cbc47547c#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485

**Comment by lippirk** [comment]
> Note that tests are not being run by travis. See [10edcba#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485](https://github.com/xapi-project/stdext/commit/10edcbabedeadf68d531b15d8ef11309d379351f#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485) and [79fc707#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485](https://github.com/xapi-project/stdext/commit/79fc7075200c10aa80ed93c5462c344cbc47547c#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485)

Good spot, I'll fix that

**Review by lindig — APPROVED** [review]

File: stdext.opam [hunk 1]
```diff
@@ -12,11 +12,9 @@ depends: [
   "ocaml"
   "dune" {build}
   "xapi-stdext-date"
-  "xapi-stdext-deprecated"
```
Comment (psafont) [inline stdext.opam:15]: I'm not sure these dependencies can be removed in the backport, I don't think they should be.

Comment (lippirk) [reply stdext.opam:15]: I agree, I backported 40ae009014bc322856a2fafb597c94140070f55a which I think I shouldn't have. I'll just do the change manually

**Comment by lippirk** [comment]
Was checking to see if travis ran the tests or not - the xapi-stdext-date opam file is wrong, but the error is ignored...

```
The following actions will be performed:

[ERROR] In the opam file for xapi-stdext-date.4.16.0:
          - At /home/opam/.opam/4.08/.opam-switch/sources/xapi-stdext-date/xapi-stdext-date.opam:10:0:
            Invalid field test
        'test' has been ignored.
  - install xapi-stdext-date 4.16.0*

[xapi-stdext-date.4.16.0] synchronised from file:///repo
```

**Comment by psafont** [comment]
Travis is fetching the env file from the master branch when it should be fetching one for stockholm.

**Review by psafont — APPROVED** [review]
I've tested this locally and the tests pass. The current issue is with travis only.

**Comment by lippirk** [comment]
Finally placated travis https://travis-ci.org/github/xapi-project/stdext/builds/753527645

**Pull request merged**